src/atoms/Button/Button.web.js

```diff
@@ -1,20 +1,471 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import styled, { useTheme } from 'styled-components';
 
-const Button = ({ type, onClick, children, ...rest }) => (
-  <button onClick={onClick} type={type} {...rest}>
-    {children}
-  </button>
-);
+import Flex from '../Flex';
+import Size from '../Size';
+import Space from '../Space';
+import Icon from '../Icon';
+import Text from '../Text';
+import View from '../View';
+import automation from '../../_helpers/automation-attributes';
+import { getColor, makePxValue, getVariantColorKeys } from '../../_helpers/theme';
+
+const BORDER_WIDTH = makePxValue(0.125);
+
+const relativePadding = (variant, padding, border = BORDER_WIDTH) => {
+  if (variant === 'tertiary') {
+    return makePxValue(padding);
+  }
+  return makePxValue(`${parseFloat(padding) - parseFloat(border)}px`);
+};
+
+const styles = {
+  fontColor({ variant, variantColor, disabled }) {
+    switch (variant) {
+      case 'primary':
+        if (disabled) {
+          return 'light.950';
+        }
+        return 'light.900';
```

Comment:
Won't this be based on `variantColor`?

Reply:
not sure about this. While other variants like `secondary`/`tertiary` were using colors like `primary.800`, the main primary button is using color in light as font is white.